packages/gatsby/src/query/file-parser.ts

```diff
@@ -135,32 +176,40 @@ async function parseToAst(filePath, fileStr, { parentSpan, addError } = {}) {
   return ast
 }
 
-const warnForGlobalTag = file =>
+const warnForGlobalTag = (file: string): void =>
   report.warn(
     `Using the global \`graphql\` tag is deprecated, and will not be supported in v3.\n` +
       `Import it instead like:  import { graphql } from 'gatsby' in file:\n` +
       file
   )
 
-type GraphQLDocumentInFile = {
-  filePath: string,
-  doc: DocumentNode,
-  templateLoc: string,
-  text: string,
-  hash: string,
-  isHook: boolean,
-  isStaticQuery: boolean,
+export interface IGraphQLDocumentInFile {
+  filePath: string
+  doc: DocumentNode
+  templateLoc: string
+  text: string
+  hash: string
```

Comment:
If you look at the return value of `getGraphQLTag`, you'll see that the hash is actually a `number`, so this is giving an error at https://github.com/gatsbyjs/gatsby/pull/27348/files#diff-b6b2b47ebc2093a46cc8cb51be3f6e9ff22d05556dd1b9139c7f83565d800200R266